dashboard/app/models/script_level.rb

```diff
@@ -556,4 +560,39 @@ def hidden_for_section?(section_id)
   def should_hide_survey(user, viewed_user)
     anonymous? && user.try(:teacher?) && !viewed_user.nil? && user != viewed_user
   end
+
+  def seeding_key(seed_context, use_existing_level_keys = true)
+    my_key = {
+      'script_level.level_keys': get_level_keys(seed_context, use_existing_level_keys),
+      'script_level.chapter': chapter,
+      'script_level.position': position
+    }
+
+    my_lesson = seed_context.lessons.select {|l| l.id == stage_id}.first
+    raise "No Lesson found for #{self.class}: #{my_key}, Lesson ID: #{stage_id}" unless my_lesson
+    lesson_seeding_key = my_lesson.seeding_key(seed_context)
+
+    my_key.merge!(lesson_seeding_key) {|key, _, _| raise "Duplicate key when generating seeding_key: #{key}"}
+    my_key.stringify_keys
+  end
+
+  # TODO: add validation so this is always in sync?
+  def get_level_keys(seed_context, use_existing_level_keys = true)
+    return self.level_keys if use_existing_level_keys && !self.level_keys.nil_or_empty? # rubocop:disable Style/RedundantSelf
```

Comment:
Oops, I see this now and think this explains it well enough: https://github.com/code-dot-org/code-dot-org/blob/dd3a5c75f9b504fdf5fa56f32da40ebf81b74597/dashboard/lib/services/script_seed.rb#L305-L309

Reply:
There are comments in `ScriptLevelSerializer` already. Added a few more on `seeding_key` in this file.

This is a little messy, and I'm not sure it's totally needed. It shouldn't be needed if the `level_keys` property (if present) is always in sync with the actual level keys. The only way they would get out of sync would be if either that property or the levels were modified not through this seeding process. But I added it to avoid bugs in case that happens.

We should also be able to remove this if we add the seeding_key GUID to ScriptLevel.

Reply:
> The only way they would get out of sync would be if either that property or the levels were modified not through this seeding process.

does that mean this could get out of sync when a levelbuilder makes edits? (I guess this might not be necessary to answer if we are planning to add the seeding_key GUID to ScriptLevel)

> We should also be able to remove this if we add the seeding_key GUID to ScriptLevel

ok, let's just plan on doing that, then. the list of things that "add the seeding_key GUID to ScriptLevel" will simplify is growing long, which is great :-)

Reply:
> does that mean this could get out of sync when a levelbuilder makes edits?

It would, but that's ok since it'd get in sync when we serialize, because of `use_existing_level_keys: false`. 

The worry would be if things were changed in the downstream environment by some process aside from seeding.